src/DonationForms/AsyncData/resources/loadAsyncData.js

```diff
@@ -0,0 +1,243 @@
+document.addEventListener('DOMContentLoaded', () => {
```

Comment:
This whole thing is impressive! can you add doc blocks with comments/descriptions to describe what's going on here?

Reply:
Resolved in b78516e3c5f33df00e0c6ff2be70dde8a51d8bf7